budgie-fuzzyclock/src/FuzzyClockApplet.vala

```diff
@@ -108,50 +127,69 @@ public class FuzzyClockApplet : Budgie.Applet
     //              This is the fun part of fuzzy-clock, feel free to
     //              be inventive within your language
-    protected string[] rules = {
+    // the format rules are divided into 12 buckets
+    // each bucket contains a rule for displaying the time within an hour
+    // This presents a problem for some languages, where it is more natural
+    // to reference the future hour numeral
+    // English Example:
+    //       "quarter after one"  --> "1:15"
+    //       "half-past one"      --> "1:30"
+    //       "quarter til two"    --> "2:45"
+    //  To satisfy the need for a future hour each rule can include a 'forward-hour offset'
+    //  the english is provided as the default, but any language can change the offset to fit
+    //  by using "<language-text> %+s" to indicate this rule needs use of the forward hour
+
+    protected FuzzyClockRule[] rules = {
         // TRANSLATORS: times between (12:58:00 - 1:02:00) are 'one-ish'
-        _("%s-ish"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("%s-ish"), 0),
 
         // TRANSLATORS: times between (1:03:00 - 1:07:00) are 'a bit past one'
-        _("a bit past %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("a bit past %s"), 0),
 
         // TRANSLATORS: times between (1:08:00 - 1:12:00) are 'ten past one'
-        _("ten past %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("ten past %s"), 0),
 
         // TRANSLATORS: times between (1:13:00 - 1:17:00) are 'quarter after one'
-        _("quarter after %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("quarter after %s"), 0),
 
         // TRANSLATORS: times between (1:18:00 - 1:22:00) are 'twenty past one'
-        _("twenty past %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("twenty past %s"), 0),
 
         // TRANSLATORS: times between (1:23:00 - 1:27:00) are 'almost half-past one'
-        _("almost half-past %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("almost half-past %s"), 0),
 
         // TRANSLATORS: times between (1:28:00 - 1:32:00) are 'half-past one'
-        _("half-past %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("half-past %s"), 0),
 
         // TRANSLATORS: times between (1:33:00 - 1:37:00) are 'twenty-five 'til two'
-        // WARNING -- the hour now reflects the upcoming hour, not the current hour
-        _("twenty-five 'til %s"),
+        // by using %+s, you can influence the hour replacement forward one hour
+        new FuzzyClockRule(_("twenty-five 'til %s"), 1),
```

Comment:
by default, this rule `FuzzyClockRule(..., 1)` references the next hour because of the 1 here.  There's currently no way to influence it to the previous hour, is this important to implement?  If so, i could implement a partner "%-s" which would decrement the hour.  Let me know if anyone is looking for that in the future